File: backend/architecture/model/patient.py
from dataclasses import dataclass
from datetime import date
from typing import Optional

@dataclass(frozen=True)
class Patient:
    identity_document: str
    identity_document_expire: date
    sanitary_document: str
    sanitary_document_expire: date
    phone_number: int
    mail: str
    password: str
    identity_document_country: Optional[str] = None
    sanitary_document_country: Optional[str] = None
    name: Optional[str] = None
    surname: Optional[str] = None
    birth_date: Optional[date] = None
    birth_country: Optional[str] = None
    nationality: Optional[str] = None
    adress: Optional[str] = None
    postal_code: Optional[int] = None
    city: Optional[str] = None
    country: Optional[str] = None

File: frontend/src/app/model/patient.ts
export interface Patient {
    identity_document: string
    identity_document_expire: string
    sanitary_document: string
    sanitary_document_expire: string
    phone_number: number
    mail: string
    password: string
    identity_document_country?: string
    sanitary_document_country?: string
    name?: string
    surname?: string
    birth_date?: string
    birth_country?: string
    nationality?: string
    address?: string
    postal_code?: number
    city?: string
    country?: string
}